addons/payment_demo/models/payment_transaction.py

```diff
@@ -103,12 +103,14 @@ def _send_capture_request(self, amount_to_capture=None):
         if self.provider_code != 'demo':
             return child_capture_tx
 
+        amount_to_capture = amount_to_capture or self.amount
         notification_data = {
-            'reference': self.reference,
+            'reference': (child_capture_tx or self).reference,
             'simulated_state': 'done',
+            'amount': amount_to_capture,
             'manual_capture': True,  # Distinguish manual captures from regular one-step captures.
         }
-        self._handle_notification_data('demo', notification_data)
+        (child_capture_tx or self)._handle_notification_data('demo', notification_data)
```

Comment:
This is not wrong, and the following comment is totally subjective, so feel free to ignore it, I wouldn't mind. ^^

I personally don't like having repetitions of `(x or y)` unless there is a reason.
Wouldn't it be better to write something like:
```py
tx = child_capture_tx or self
```
And then use it for the reference, the amount, and for the `tx` calling `_handle_notification_data` ?